src/widget.cpp

```diff
@@ -242,6 +244,16 @@ void widget::load( const JsonObject &jo, const std::string & )
         for( const std::string wid : jo.get_array( "widgets" ) ) {
             _widgets.emplace_back( widget_id( wid ) );
         }
+    } else {
+        if( jo.has_object( "extend" ) ) {
+            JsonObject tmp = jo.get_object( "extend" );
+            tmp.allow_omitted_members();
+            if( tmp.has_array( "widgets" ) ) {
+                for( const std::string wid : tmp.get_array( "widgets" ) ) {
+                    _widgets.emplace_back( widget_id( wid ) );
```

Comment:
There's already 5 copies of stuff like this in `monstergenerator.cpp` and it feels like unnecessary boilerplate. Some other classes like `mutation` appear to support this without additional JSON loading code. It would be great if more than just "widgets" could be extended by mods.

Reply:
Switching this section to using `optional` with a `string_id_reader` allows "extend" to modify "widgets" with no extra code here.